Would love to see you use jupysql

idomic opened the pull request:
Hey Chang how are you??
I'd recommend using jupysql as it will give you more flexibility on visualization and integrations.

I can submit a clean PR (I couldn't open an issue so I went this route).

changhiskhan commented on src/sql/index.html:
```diff
@@ -28407,11 +28408,39 @@
-                  t(n);
-                });
+              window.IPython ||
+                fetch("./oxford_pet.json")
```
this is hard coded still

anatoly-eto replied:
It is fallback for local development.
It never gets executed in notebook.

changhiskhan commented on src/sql/index.html:
```diff
@@ -28407,11 +28408,39 @@
+                if (!window.IPython) return;
+                console.log("executing\u2026"),
+                  await window.IPython.notebook.kernel.execute(
+                    "df.to_json(None, orient='records')",
```
the variable `df` isn't always going to be the one you want.

anatoly-eto replied:
Yes-yes, we can even abstract the command to execute behind our private Python method.

changhiskhan reviewed (commented):
there's still a hard-coded oxford_pet.json

also, `df` is not the right output variable to use. I think we need to generate a unique var name and use that instead

anatoly-eto commented:
Closing because widget was already updated here:
- https://github.com/eto-ai/ipython-sql/pull/4

edublancas committed:
adds --schema/-s arg to %sqlcmd tables

edublancas committed:
updates guide

tonykploomber committed:
Added more telemetry (#89)

* Add: log_call for historgram & boxplot

* Add: log_call for historgram & boxplot

* Add: log_call for historgram & boxplot

* WIP: checking decorator method called

* Add: histogram test

* Refract: share mock_log_api as fixture

* Add: test for DataFrame

* Fix: lint

* Add: sqlrender

* Add: execute

* Fix: test boxplot fail

* Add: CHANGELOG doc

* Add: store csv to tmp folder

* Fix: add prefix to CHANGELOG

* Fix: test case

* trigger rebuild

* Change: Feature -> Fix

---------

Co-authored-by: Tony Kuo <tonykuo@Tonys-MacBook-Pro.local>

tonykploomber committed:
Add more doc details on telemetry (#101)

* Update: doc in community

* Update: order

* Add: more info

* Fix toc Order

* Fix: coc comma

edublancas committed:
adds `%sqlcmd` to list tables and schemas

edublancas committed:
Update CHANGELOG.md

edublancas committed:
sql release 0.5.3

edublancas committed:
Bumps up sql to version 0.5.4dev

yafimvo committed:
flake8 and black added to setup. cd script added.

yafimvo committed:
black dependency moved to dev

yafimvo committed:
email changed

edublancas committed:
Merge pull request #105 from yafimvo/cd_jupysql

CD for jupysql added

gtduncan committed:
updates doc links (#103)

tonykploomber committed:
Adds jupysql community messages on ValueError, TypeError (#106)

* Fix: magic_plot & its test case

* Fix: plot & its test case

* Fix: plot & its test case

* Fix: plot & its test case

* Fix: store & its test case

* Fix: inspect & its tests

* Fix: lint

* Fix: lint

* Fix: test case

* Fix: test case

* Fix: use modify_exceptions decorator

* Fix: TypeError

* Fix: lint

* Add: Change Log

* Cleanup

* Add: check TypeError

* Fix: CHANGELOG

tonykploomber committed:
Fix: Doc Typo in Connecting: PostgreSQL (#110)

* Fix: Typo

* Fix: CHANGELOG

* Remove: Changelog

edublancas committed:
adds pr template

edublancas committed:
adding `%jupysql`/`%%jupysql` as alias (#112)

* adding alias

* adds test for %jupysql alias, documents it and adds changelog

* addresses PR comments

edublancas committed:
fixes changelog

edublancas committed:
sql release 0.5.4

edublancas committed:
Bumps up sql to version 0.5.5dev

tonykploomber committed:
Integration tests for postgres and mysql (#111)

edublancas committed:
Update README.md

edublancas committed:
adds tutorial on JSON + DuckDB (#119)

edublancas committed:
Update CHANGELOG.md

edublancas committed:
better error message on failed connection (#120)

edublancas committed:
Update json.md

edublancas committed:
Update CHANGELOG.md

edublancas committed:
sql release 0.5.5

edublancas committed:
Bumps up sql to version 0.5.6dev

edublancas committed:
Update ci.yaml

tonykploomber committed:
more db integration testing (#117)

edublancas committed:
running flake8 on notebooks during CI + formatting (#125)

* running nbqa during CI, formatting notebooks

* linting in two steps

edublancas committed:
fix ci

tonykploomber committed:
clearer error message when failing to connect to the db (#127)

gtduncan committed:
dynamic binder links (#135)


---------

Co-authored-by: Eduardo Blancas <github@blancas.io>

edublancas committed:
sql release 0.5.6

edublancas committed:
Bumps up sql to version 0.5.7dev

edublancas committed:
comments _config.yml

edublancas committed:
adds example to convert to polars (#139)

* adds example to convert to polars

* updates dependencies

yafimvo committed:
fail docs if a notebook crashes (#133)

tonykploomber committed:
telemetry: logging db metadata (dialect, driver, server version) (#126)

edublancas committed:
Update setup.py

tonykploomber committed:
Fix: doc build fail cause _get_curr_connection_info (#165)

* Fix: Plot.py

* Fix: Plot.py

edublancas committed:
adds developer guide (#163)

* adds developer guide

* fix

* fix

* updates doc requirements

* updates contributing file

edublancas committed:
Update CONTRIBUTING.md

jorisroovers committed:
Support for SqlMagic.autopolars (#142)

* Support for SqlMagic.autopolars

Allows auto conversation of sql cells to polars Dataframes.

* Add changelog update

* Implement mutex behavior for autopandas and autopolars

Includes test.

* drops old IPython imports

---------

Co-authored-by: Eduardo Blancas <github@blancas.io>

edublancas committed:
listing incompatibilities with ipython-sql (#169)

* updates changelog

* adds section on ipython-sql incompatibilities

edublancas committed:
sql release 0.6.0

edublancas committed:
Bumps up sql to version 0.6.1dev

edublancas committed:
pins jupyter-book<0.14 (#181)

* ci

* pins jupyter book

tonykploomber committed:
Add: test - error when there is no active connection (#171) (#189)

* Add: test

* Fix: pytest

* Revert

* Revert

* Move: _get_curr_connection_info to classmethod level

* Move: pytest fixture cleanup

* pins jupyter-book<0.14 (#181)

* ci

* pins jupyter book

* Add: f-string fix

* Fix: arg self -> cls

* Fix: arg self -> cls

---------

Co-authored-by: Eduardo Blancas <github@blancas.io>

Palashio committed:
upgrades for prettytable (#191)

tonykploomber committed:
Adds support for parameterizing queries using `{{variable}}` (#137)

edublancas committed:
clearer error message (#193)

edublancas committed:
sql release 0.6.1

edublancas committed:
Bumps up sql to version 0.6.2dev

ayushmaanshrotriya committed:
improves duckdb tutorial (#190)

edublancas committed:
Syntax highlighting tutorial (#196)

edublancas committed:
adds ms sql server tutorial (#200)

edublancas committed:
adds polars tutorial (#202)

edublancas committed:
mysql tutorial (#204)

edublancas committed:
adds html metadata to all doc sections (#207)

edublancas committed:
adds mariadb tutorial (#208)

* adds mariadb tutorial

* removes old section

edublancas committed:
adds clickhouse tutorial (#209)

edublancas committed:
migration to jupyter book 0.14 (#211)

* unpins jupyterbook

* migrating jupyter book

* fixes

* fixes

edublancas committed:
enabling htmlzip docs (#212)

edublancas committed:
removes htmlzip from docs config

edublancas committed:
fixes docs image

tonykploomber committed:
Fix: deprecation warning incorrectly displayed (#213)

edublancas committed:
adds instructions to supress warnings

edublancas committed:
adds link to supress warnings

edublancas committed:
sql release 0.6.2

edublancas committed:
Bumps up sql to version 0.6.3dev

idomic committed:
Updating license (#198)

* Updating license

* Updating license

tonykploomber committed:
Add: button & get started image (#216)

tonykploomber committed:
Fix - Issue better detect the user uses sqlalchemy variable expansion (#215)

* Add: fail test case

* Add: better check for variable usage

edublancas committed:
makes warning condition more efficient

edublancas committed:
sql release 0.6.3

edublancas committed:
Bumps up sql to version 0.6.4dev

gtduncan committed:
fix `invoke doc` command (#192)

neelasha23 committed:
jupysql vs ipython-sql (#225)

Palashio committed:
support for sqlalchemy 2.0 (#219)

edublancas committed:
fixes changelog

edublancas committed:
adds test for checking dollar sign in column name (#231)

jpjon committed:
updated to include listing tables & columns (#227)

edublancas committed:
sql release 0.6.4

edublancas committed:
Bumps up sql to version 0.6.5dev

yafimvo committed:
using sqlalchemy autocommit when possible (#91)

edublancas committed:
fix changelog

edublancas committed:
changes title to tutorial

gtduncan committed:
improvements to json tutorial (#222)

yafimvo committed:
ignoring binder.ploomber.io when checking broken links (#252)

gtduncan committed:
analyzing data from github api tutorial (#239)

ayushmaanshrotriya committed:
Invoke setup on Windows fails fixed (#241)

* Invoke setup on Windows fails fixed

* Delete tempCodeRunnerFile.py

* black

---------

Co-authored-by: Eduardo Blancas <github@blancas.io>

edublancas committed:
Delete .DS_Store

edublancas committed:
Update .gitignore

tonykploomber committed:
changes printed messages for logger (#254)

Palashio committed:
future warning for `--with` and `--save` (#249)



---------

Co-authored-by: Eduardo Blancas <github@blancas.io>

edublancas committed:
sql release 0.6.5

edublancas committed:
Bumps up sql to version 0.6.6dev

edublancas committed:
updates docs logo

tonykploomber committed:
Deprecate sql parametrizing behavior (#237)

* Update: doc

* Drop features

* Remove: test case

* Remove: test case

* Revert: duckdb.md

* Change: to logger debug

* Revert "Change: to logger debug"

This reverts commit 9840710e14fdc4840050453fa46d450f60717189.

edublancas committed:
pin sqlalchemy 1.x (#261)

* pins sqlalchemy 1.x

* sql release 0.6.6

* Bumps up sql to version 0.6.7dev

edublancas committed:
bumps up version

DaveOkpare committed:
Refactor run function and added unit test (#267)

tonykploomber committed:
adds missing test case (#238)

* Add: test cases

* Add: test cases

* Update more cases

* Update: test cases

* Add: more bad query test cases

* Downgrade: sqlalchmey

---------

Co-authored-by: Eduardo Blancas <github@blancas.io>

Palashio committed:
217 Add sql cmd for test (#266)

* changes

* lint

* made integration test modifications

* fixes, and integration test additions

* remove print

* edit md

* change print statement

* change text

edublancas committed:
CI updates (#270)

* updates pyproject.toml

* deletes tasks.py

* using pkgmt lint for linting

* adds missing metadata filter

* formatting with black

idomic committed:
Update README.md

yafimvo committed:
add `%sqlcmd profile` (#168)

* table profile added

* lint

* test fixed

* lint

* autopolars property added to config

* save report added

* percentile_disc added, schema added, docs updated

* numpy added to setup

* np removed, run_raw added, queries updated, test fixed

* test fixed

* config.autolimit check fixed

* integration tests added

* integration tests fixed

* lint

* index removed from integration tests

* postgres, mysql and maria excluded from profile test

* lint

* postgresql fixed

* postgresql nan values fixed

* rebase

* naming changed

* sqlalchemy downgraded to 1

* config removed from raw_run

tonykploomber committed:
using sqlglot for better SQL generation (#164)

idomic committed:
CI badges (#295)

* Initial commit

* Adding integration tests and broken-links

idomic committed:
badge edits

AnirudhVIyer committed:
clickable links in ResultSet (#274)

edublancas committed:
Update CHANGELOG.md

gtduncan committed:
postgres tutorial improvements (#259)

yafimvo committed:
ggplot plotting API (#226)

edublancas committed:
adds notes on unreleased features (#299)

edublancas committed:
updates pr template

yafimvo committed:
ggplot telemetry (#303)

tonykploomber committed:
interactive parametrized SQL queries (#293)

anupam-tiwari committed:
clearer error when passing `--with` with a missing name (#290)

* fixed 257-better-error-when-missing-CTE with added test cases

* fix

* updates changelog

---------

Co-authored-by: Anupam <anupam@10-16-252-145.dynapool.wireless.nyu.edu>
Co-authored-by: Eduardo Blancas <github@blancas.io>

tonykploomber committed:
Adding integration tests locally via docker (#281)

* WIP

* Try to push

* Lib

* fix

* add: environment

* fix yield issue

* move var

* fix return yield

* fix return yield

* add: test_profile_query

* add: test_profile_query

* Move databaseconfighelper to _testing

* Cleanup

* Add: tests

* Update: test

* fix: lint

* fix

* Revert

* Revert: setup.py

* Fix: test cases

* Add: DatabaseConfigHelper test cases

* Fix: setup

* Fix: setup

idomic committed:
ETLs markdown guide (#306)

* new guide

* Added to toc

* Added header + removes integration headline from guides

* Review fixes

* Removing header

* Added blog goals + runnable output

* Added blog goals + runnable output

* Added blog goals + runnable output

* locking polars

edublancas committed:
improves docs on connection alias (#315)

chuwpeng123 committed:
documenting jupyter magics in docs (#240)

gtduncan committed:
Disabled autocommit for pytds (#308)

* pytds autocommit

* lint

* changelog and test case

* lint

* added warning check

added warning check

added warning check

tonykploomber committed:
Fix mssql.ipynb typo (#326)

`pip install pyodbc=4.0.34` isn't valid command

Should be `pip install pyodbc=4.0.34`

raghavSharmaCode committed:
Better exception handling for '%sqlcmd'. (#271)

* updated magic_cmd.py

* updated magic_cmd.py

* updated magic_cmd.py

* updated magic_cmd.py to raise Usage Error for empty command string

* updated magic_cmd.py to raise appropriate Usage Error

* Added validation checks for %sqlcmd commands

* updated magic_cmd.py

* resolved conflicts

* Update src/sql/magic_cmd.py

Co-authored-by: Tony Kuo <123580782+tonykploomber@users.noreply.github.com>

* Update src/sql/magic_cmd.py as per suggestion

Co-authored-by: Tony Kuo <123580782+tonykploomber@users.noreply.github.com>

* fixed test cases

* fixed lint errors

* resolved lint errors

* Added method description. Also, added a few more test cases of %sqlcmd input

* Update CHANGELOG.md

---------

Co-authored-by: Tony Kuo <123580782+tonykploomber@users.noreply.github.com>
Co-authored-by: Ido M <michael.ido@gmail.com>

lfunderburk committed:
CTE tutorial improvements (#300)

mehtamohit013 committed:
Adds support for MSSQL  (#320)

* Adds support for MSSQL  (#320)

---------

Co-authored-by: Tony Kuo <yen.c@ploomber.io>

edublancas committed:
Update CONTRIBUTING.md

edublancas committed:
Update environment.yml

neelasha23 committed:
SQL Autocompletion (#323)

* Autocompletion

---------

Co-authored-by: Eduardo Blancas <github@blancas.io>

chuwpeng123 committed:
moves csv tutorial (#322)

yafimvo committed:
Error message when user passes a wrong table/schema name (#324)

* error message added

* with added, tests fixed

* test fixed

* lint and chanelog

* Support table name with star

* safe _is_table_exists method added

* test added, doctsring added

* tests order changed

* test_no_active_session updated

* _is_table_exists fixed

* lint

* default identifiers added

* default identifiers fixed

* func annotations added, f-string fixed

* constants added

---------

Co-authored-by: Ido M <michael.ido@gmail.com>
Co-authored-by: Eduardo Blancas <github@blancas.io>

edublancas committed:
adds two workflows (#343)

jorisroovers committed:
Support for SqlMagic.polars_dataframe_kwargs (#325)

* Support for SqlMagic.polars_dataframe_kwargs

Allows for passing of custom keyword arguments to the Polars DataFrame
constructor.

Fixes #312

* Improvements for SqlMagic.polars_dataframe_kwargs

- Updated docs
- Fixed failing tests
- Passing polars_dataframe_kwargs as kwargs instead of dict

* Black formatting fixes

chuwpeng123 committed:
update md headers (#340)

Co-authored-by: lvdousha <61637161+lvdousha@users.noreply.github.com>

edublancas committed:
only check changelog when review has been requested

edublancas committed:
installing doc dependencies from lockfile (#345)

edublancas committed:
updates changelog.md

edublancas committed:
adds missing links to changelog.md

edublancas committed:
applies black formatting

tonykploomber committed:
Fix: lock pgspecial package version to 2.0.1 (#349)

edublancas committed:
changing event that triggers changelog check

edublancas committed:
removes changelog.md workflow

jng9134 committed:
update %sqlrender api doc (#279)

* update %sqlrender api doc

* Update magic-plot.md

* Update magic-render.md

* Update magic-sql.md

* Update magic-render.md

* Update magic-render.md

* Update magic-plot.md

* Update magic-render.md

* Update magic-render.md

* Update magic-render.md

changed variable names for clearer meaning

* Update CHANGELOG.md

---------

Co-authored-by: Ido M <michael.ido@gmail.com>

neelasha23 committed:
CI Shortening (#332)

* CI Shortening

actions role

aws install

aws install

aws install

aws install

aws install

aws install

aws install

aws install

added assumerole

assumerole

Revert ci changes

Empty commit

cache upload

cache upload

cache upload

cache upload

folder

ls

ls

changed dir

prebuild

Empty commit

git id

git id

git id

version

Empty commit

removed logs

Empty commit

awscli dep

lock file

lock file

revert

revert

Removed access keys;

testing github actions

testing github actions

testing github actions

testing github actions

testing github actions

testing github actions

testing github actions

revert ci changes

revert ci changes

* Empty commit

neelasha23 committed:
Parallel tests (#352)

* Parallel tests

* Parallel tests

tonykploomber committed:
Drop sqlglot for generating CTEs (#318)

edublancas committed:
updates changelog

edublancas committed:
updates integration tutorials (#356)

* updates integration tutorials

* lint

edublancas committed:
creates integration testing dependencies (#359)

edublancas committed:
disable parallel tests (#363)

yafimvo committed:
more efficient table check, better error if user possibly missing `--with` argument (#361)

* suggestion for stored query added

* checks if table in store when passing --with

* lint

edublancas committed:
fixing problems with comments that have `%` and fixing incompatibility with MySQL (#362)

edublancas committed:
updates changelog

edublancas committed:
updates admonitions (#364)

* updates admonitions

* adds pip command to upgrade

edublancas committed:
sql release 0.7.0

edublancas committed:
Bumps up sql to version 0.7.1dev

DaveOkpare committed:
Update connecting.md (#272)

* Update connecting.md

* Added tip to dsn connections docs

* Updating connecting guide (#56)

* Update CHANGELOG.md

---------

Co-authored-by: Ido M <michael.ido@gmail.com>

edublancas committed:
syntax highlighting using jupysql-plugin (#377)

edublancas committed:
adds conda install command to quickstart (#376)

* adds conda install command to quickstart

* updates changelog and readme

edublancas committed:
fixes duplicated ci jobs

edublancas committed:
adds compat table (#378)

* adds compat table

* adds postgres

* adds snowflake

* adds the rest of the databases

edublancas committed:
Integration testing fix (#375)

tonykploomber committed:
Add sqlalchemy 2 compatibility (#339)

* Fix: partially plot

* Fix: some test bug

* Enable: 2

* Fix: partial inspect

* Fix: error inspect

* Fix: magic.py

* fix: magic_cmd

* fix cmd

* Fix: parse

* update ci

* fix inspect

* Reproduce issue 1

* fix session commit

* Fix: test plot

* Use ip to connect

* disable inspect

* Skip in 3.7

* Lint

* Update: doc build

* Bring back fail test_inspect

* Fix: test_inspect

* Revert: inspect.py

* Clean up

* Clean up

* remove: python 3.7

* Add: doc

* Fix: lint

* fix: comments

* Update pandas.md

---------

Co-authored-by: Ido M <michael.ido@gmail.com>

edublancas committed:
documenting how to run integration tests locally (#380)

* documenting how to run integration tests locally

* fixes

edublancas committed:
fixes branch name in ci settings

tonykploomber committed:
Snowflake integration tests (#336)

* Add: Init

* setup secret

* Fix: table hash name & refract

* revert doc/api/magic-plot.md

* Fix: postgre

* fix commit

* fix load_generic_testing_data

* Update: integration test

* Fix: typo

* Fix test fail

* fix test

* Fix: ci fail

* Fix comment

* Fix: test

* fix: test

* fix: test_sqlcmd_tables_columns

* fix mssql

* fix: test_sqlcmd_tables_columns

* Small issue

* Rebase

tonykploomber committed:
disable snowflake (#393)

edublancas committed:
Update pull_request_template.md

tonykploomber committed:
transpile query not classmethod (#389)

* transpile query not classmethod (#389)

edublancas committed:
fix `%sqlcmd tables/columns` in MySQL (#370)

* fixing %sqlcmd tables/columns in MySQL

* updates mariadb and mysql tutorials

* fixes

* fix

* adds test with custom schema

lfunderburk committed:
Improve connecting tutorial (#385)

edublancas committed:
adds workflow-edits workflow

edublancas committed:
fixes branch name

edublancas committed:
updates documentation lock file (#404)

* updates lock file

* updates lock file

* switching to non-lock file

* update lock file

* fix

tonykploomber committed:
Add remote integration tests flow (#396)

* dummy text

* Fix: environ

* fix ci file

* Revert "fix ci file"

This reverts commit ab64caa16c7a4642aebfc1da0a9ab68aa1776f74.

* Revert "Fix: environ"

This reverts commit 7c4694dfe8ad866f2df6cbf98501ffb727aebb9b.

* Try: pull_request_target

* revert

* add: individual CI

* update: ci file

* fix fixture

* fix live ci file

* fix ci

* fix ci

* fix ci

* fix ci

* fix ci

* fix ci

* Move init

* fix ci

* fix conftest

* fix lint

* Add: conditional exit

* fix conditional

* fix syntax

* fix status code

* try conditionally

* fix syntax

* run steps conditionally

* fix flow

* just echo

* echo

* github.base_ref

* another way

* fix

* update

* fix ci

* fix

* test

* fix

* fix

* test early return

* Revert "fix"

This reverts commit 87c8061378f4ffdd656119a9653860e5704913e1.

* Revert "test early return"

This reverts commit 5ded16ab35a3ed6a969e21529b0571be751d2833.

* better flow

* add doc

* fix

* typo

* Fix doc

* Add: abstract on skip_on_live_mode & skip_on_local_mode

* fix

* Rename

* fix: text

edublancas committed:
adds more methods for connecting to a db (#399)

* tutorial improvements

* improvements

* fix

* fixes

mehtamohit013 committed:
Fix polars version (#408)

* Testing

* Fix

* Check

yafimvo committed:
Add algolia search (#351)

* algolia search added

* changelog

* index updated

* algolia v3 added

* lint

* typo and css style

* reinvoke docs build

idomic committed:
Fixing polars version (#412)

gtduncan committed:
%sqlcmd improvements (#341)

* %sqlcmd improvements (#341)

idomic committed:
commenting snowflake. (#413)

edublancas committed:
sql release 0.7.1

edublancas committed:
Bumps up sql to version 0.7.2dev

mehtamohit013 committed:
Fixing sqlcmd tests (#415)

tonykploomber committed:
Improve boxplot SQL perf - Part 2 (#152)

* Add: fix

* Add: test case

* Add: fail test

* rebase

* cleanup

* Constrain sqlglot version

* Revert

* print msg

* lint

* more print

* Revert: print

* Clean up dialect

* update: test_plot

* Add: flatten

* Fix: example

* Add: flatten

* Fix: lint

* fix example

* Add: CHANGELOG

* Try: master branch connection

* Try: master branch connection

* Add: test case

* fix: lint

yafimvo committed:
support DB API 2 drivers (#350)

* ggplot and histogram added

sql added

sqlcmd

lint

* lint

* lint

* lint

* custom connection wrapper added

* lint

* tasks deleted

* custom connection fixed after rebase

* lint

* sqlalchemy 2 compatibility added

* lint

* sqlalchemy<2

* setup rolled back

* sqlalchemy2 compatibility added

* CI fixed

* integration test added

* lint

* CI fixed, tests added

* CI fixed, docs and ggplot tests added

* docs fixed

* CI fixed (sqlalchemy 2)

* rejects fixed, more tests added, support_only_sql_alchemy_connection added

* telemetry added

* questdb example added to docs

* lint

* custom-connection example updated

* CI fixed

* connecting.md file fixed

* dp-api.ipynb updated

* timeout added to db-api

* timeout removed

* conf.py updated

* cleaned, classmethod removed

* fetch results with custom connection fixed

* example fixed

* is_custom_connection fixed

* doc updated

* lint

* lint

* docs updated

* _transpile_query removed on custom connection

* integration tests fixed

* questdb tutorial updated

edublancas committed:
Update CHANGELOG.md

edublancas committed:
Add custom errors with no traceback (#422)

* adds custom error with no traceback

* test fixes

* fix

* adds exceptions section to dev docs

* changing some other errors

* fix

* adds clarification note

* changelog updates

* changes due to review

edublancas committed:
fixes %sqlcmd table, plot when column or table has spaces (#424)

* fixes %sqlcmd plot when column or table has spaces

* sanitizing arguments in %sqlcmd columns

* xfailing some tests on windows

idomic committed:
Adding snowflake back!

edublancas committed:
Update CHANGELOG.md

tonykploomber committed:
Sticky column when showing profile results  (#411)

* add: css

* fix: doc

* fix doc

* fix

* fix test & highlight

* Fix: table css

* fix css

* fix: lint

tonykploomber committed:
document how to generate vendor-agnostic SQL with sqlglot (#371)

* Add: doc

* clean

* Fix: static

* WIP

* fix: doc

* fix: lint

* revert kernel

* fix: doc cells

* fix: doc

* fix: metadata

* add: css

* Revert "add: css"

This reverts commit 18d22944ff7a8d6f9fa16c6203ece68031568de9.

* Fix: plus

* Revert "Fix: plus"

This reverts commit 1ac4d5041bd83a2215a64ccb50c529e89c654ab7.

* fix: plus

edublancas committed:
migrating more error tracebacks (#431)

edublancas committed:
adds workflow for uploading to PyPI

edublancas committed:
Update CHANGELOG.md

edublancas committed:
sql release 0.7.2

edublancas committed:
Bumps up sql to version 0.7.3dev

javier committed:
(docs): Typo on JupySQL (#454)

JupySQL was incorrectly shown as JuppySQL with double 'p'. Fixed

mehtamohit013 committed:
SQL with Excel (#423)

* how to excel

* Add df

* Moved

* Added to _toc

* Fix CI

* DuckDB and Pandas

* Updated Changelog

* Formatted

* Fix DuckDB excel

* Fix

edublancas committed:
fix `--alias` when passing an existing engine (#446)

mehtamohit013 committed:
Fix excel tutorial link (#464)

idomic committed:
Adding addmonitions to excel guide

idomic committed:
Fixing ipython version to 8.12 on python 3.8 (#466)

* test commit

* locking ipython to 8.12.0

* locking ipython to 8.11.0

* locking ipython to 8.12.0

* locking ipython to 8.12.0 to py3.8

idomic committed:
updating changelog entry for ipython fix

idomic committed:
sql release 0.7.3

idomic committed:
Bumps up sql to version 0.7.4dev

idomic committed:
removing unwanted files.

idomic committed:
Adding changelog entries

idomic committed:
sql release 0.7.4

idomic committed:
Bumps up sql to version 0.7.5dev

idomic committed:
285 doc tables columns (#465)

* #285-doc-tables-columns (#394)

* #285-doc-tables-columns

* Update 285

* Update magic-tables-columns.md

---------

Co-authored-by: lvdousha <61637161+lvdousha@users.noreply.github.com>
Co-authored-by: Ido M <michael.ido@gmail.com>

* Cleaning md

* review changes

---------

Co-authored-by: chuwpeng123 <117867769+chuwpeng123@users.noreply.github.com>
Co-authored-by: lvdousha <61637161+lvdousha@users.noreply.github.com>

idomic committed:
documenting `%sqlcmd tables`/`%sqlcmd columns` changelog entry

edublancas committed:
Update setup.py (#471)

edublancas committed:
improving duckdb + autopandas performance  (#469)

* Using native DuckDB `.df()` method when using `autopandas`

* lint

* test fixes

* adds polars to the benchmark

* adds missing unit tests for ResultSet

* ResultSet no longer a subclass of list

* fix

* ci

* ci

* ci

* fix

* checking result set with non sqlalchemy

* ci

* fixes notebook

edublancas committed:
fixes CD workflow (#472)

* Update ci.yaml

* ci

* Update ci.yaml

---------

Co-authored-by: Ido M <michael.ido@gmail.com>

tonykploomber committed:
Ensure CI integration testing is using sqlalchemy2 (#482)

* update: config

* rebuild

* rebuild

* update version

* Revert "update version"

This reverts commit 5db6166a550180641610ca263bff6599a7b98d6b.

* why not working

* update

* fix syntax error

* Fix syntax

* Fix version

* revert

* revert 2

* add print on local

* update local

* revert setup.py

* Fix version

* Revert "Fix version"

This reverts commit 667f72eede5901b965185385679bc32a16fd3451.

* add cloud

* update live version

* update to -U

* lock sqlalchemy version

neelasha23 committed:
Plot error message (#441)

* success case

* Plot error message

testcase

Fix

Fix

Fix

test fix

test fix

test fix

changelog

usageexception

test

test

error

error

error

error

error

changed to runtimeerror

quest db test

Revert test

Revert test

exception type test

Review comments

changelog

rebase

changelog

Review comments

changed to sqlite

change sequence

lint

revert

added tmp_empty

failure case

tonykploomber committed:
Change default displaylimit value (#462)

* WIP

* add: unlimit test case

* Update: default value

* Fix: lint

* fix lint

* fix: merge error

* revert

* fix lint

* Add: doc

* add validator

* fix: lint

* fix changelog

* update: vs.md

* fix: default none

* add: no limit test case

* fix setup

* fix lint

* add: test_displaylimit_with_limit_clause

* add test cases

* igonre this line

* lint issue

* Update doc

* trigger rebuild

* rebuild - 1

* change optionsss

* update configuration

* python

* Update

neelasha23 committed:
SQLachemy error overriding (#400)

SQLachemy error overriding (#400)

tonykploomber committed:
Fix CI integration live issue (#493)

* add pipeline tag

* rebuild

* fix snowflake-sqlalchemy version

* old snowflake-connector-python==3.0.2

* lint

* downgrade sqlalchemy 2

* only run on test_generic_db_operations

* test_generic_db_operations

* add skip on questDB

* fix non live db

* rebuild

* remove connector version

* downgrade duckdb version

* rebuild

* fix version

* revert version

* rebuild

* rebuild

* comment error

* downgrade duckdb

* add back

* comment out alias

* lint

* suggestion

* disable two test cases

* disable two

* fix lint

* rebuild

* ignore questdb

* single statement

* rebuild

* exclude generic operation

* add mssql time

* comment tables& columns

* skip mssql

* skip mysql and mariadb

* revert questDB

* try to close sessions

* test new way to drop table

* fix lint & see logs

* add dispose

* disable test_custom_connection_error

* clear print

* remove profile

* remove profile

* cleanup code

* lint

* comment -> skip

* fix lint

* fix lint

* rebuild

* ci-integration-db

* add rest tests

* mysql

* fix: comments

neelasha23 committed:
pretty table fix (#500)

neelasha23 committed:
CI fix (#496)

mehtamohit013 committed:
raise exception if SQLQuery contains hyphens (#504)

edublancas committed:
adds sql formatting guide (#473)

* adds sql formatting guide

* adds jupysql-plugin to binder env

* adds instructions to launch notebook to format sql

* updates env

* changes format-sql format from md to ipynb

neelasha23 committed:
syntax error test fix (#509)

edublancas committed:
sql release 0.7.5

edublancas committed:
Bumps up sql to version 0.7.6dev

neelasha23 committed:
Product analytics tutorial (#485)

* Product analytics tutorial

toc

fix

tutorial

fixed test

review comments

redundant newline

* Empty commit

tonykploomber committed:
Oracle Integration testing (#477)

TegveerG committed:
trinodb integration guide first review  (#516)

* trinodb integration guide first review

* fix linting  errors in PR

* linting error fixed (checked locally)

* jupysql env config then ran pkgmt to fix linting

* run docker locally and update conf.py so readthedocs wont run docker

* fix readdocs: added trinodb to toc

---------

Co-authored-by: Eduardo Blancas <github@blancas.io>

mehtamohit013 committed:
showing integration guide if connection fails (#503)

* Added db checking

* Test Fix

tonykploomber committed:
CD script not taking into account integration tests (#520)

jorisroovers committed:
Syntactic sugar for outputting result variables (#501)

edublancas committed:
adds `codespell` (#527)

tonykploomber committed:
--persist-replace argument (#440)

edublancas committed:
fix typos (#529)

mehtamohit013 committed:
checking custom connections according to PEP 249 (#517)

* Added PEP 249 compilance

* Changelog

* Added test

* Fix tests

* Fix tests

* Update CHANGELOG.md

Co-authored-by: Eduardo Blancas <edu.blancas@gmail.com>

* Update src/sql/connection.py

Co-authored-by: Tony Kuo <123580782+tonykploomber@users.noreply.github.com>

* Added test case

* Format and lint

---------

Co-authored-by: Eduardo Blancas <edu.blancas@gmail.com>
Co-authored-by: Tony Kuo <123580782+tonykploomber@users.noreply.github.com>

yafimvo committed:
add `%sqlcmd explore` (#420)

* initial version added

* sql pagination added

* test fixed

* sort columns added

* changelog updated

* Allows to add multiple tables to a single notebook

* tests added

* testing on binder

* binder test

* _get_binder_hub_url added

* webscoket connection added

* table_widget added

* rebase

* lint

* widgets directory name changed

* imports

* short import removed

* __init__ added

* import fixed

* lint

* import

* telemetry added

* pagination bug on Jupyter nb fixed

* docs added

* clean

* lint

* tests added

* path fixed

* path changed

* lint

* table_widget js and css files added

* empty commit invoke CI

* apt.txt added with nodejs

* apt removed. doc updated

* apt removed

* docs fixed

* apt.txt added

* docs

* cleaned

* code review fixes

* code review fixes

* lint

* lint

* psutil added

* lint

* link to binder fixed

* table_widget.jpg removed

edublancas committed:
fixes codespell

yafimvo committed:
documenting different connection types (#515)

* docs updated, native connections removed

* changelog

* lint

* lint

* CustomConnection documented

---------

Co-authored-by: Eduardo Blancas <github@blancas.io>

tonykploomber committed:
Fix CI integration local issue (#535)

* add pipeline symbol

* remove oracle db in test_sqlplot_histogram

* add test_sqlplot_boxplot in oracle

* test_sqlplot_boxplot

edublancas committed:
sql release 0.7.6

edublancas committed:
Bumps up sql to version 0.7.7dev

edublancas committed:
adds missing install cmd

idomic commented:
Closing this as it was a PR from master which is way behind.
